Add Container Linux kubeadm installing instruct

klausenbusk opened the pull request:
Tested on a DigitalOcean 2GB droplet per the doc and:
```
sudo kubeadm init --pod-network-cidr=10.244.0.0/16
sudo kubectl --kubeconfig=/etc/kubernetes/admin.conf apply -f https://raw.githubusercontent.com/coreos/flannel/v0.9.1/Documentation/kube-flannel.yml
```


cc @luxas

See also: https://github.com/kubernetes/kubeadm/issues/300

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/website/6768)
<!-- Reviewable:end -->

luxas commented on docs/setup/independent/install-kubeadm.md:
```diff
@@ -19,6 +19,7 @@ see the [Using kubeadm to Create a Cluster](/docs/setup/independent/create-clust
   - RHEL 7
   - Fedora 25/26 (best-effort)
   - HypriotOS v1.0.1+
+  - Container Linux
```
Please document what CoreOS version you used to test this with...

luxas commented on docs/setup/independent/install-kubeadm.md:
````diff
@@ -220,9 +231,44 @@ systemctl enable kubelet && systemctl start kubelet
+Install `kubeadm`, `kubelet`, `kubectl` and add a `kubelet` systemd service:
+
+```bash
+RELEASE="$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)"
````
nit: `dl.k8s.io/release/stable.txt` is a better shorthand for this

klausenbusk replied:
Done

luxas commented on docs/setup/independent/install-kubeadm.md:
```diff
@@ -220,9 +231,44 @@ systemctl enable kubelet && systemctl start kubelet
+chmod +x {kubeadm,kubelet,kubectl}
+
+cd "/etc/systemd/system/"
+curl -L "https://raw.githubusercontent.com/kubernetes/kubernetes/${RELEASE}/build/debs/kubelet.service" | sed 's:/usr/bin:/opt/bin:g' > kubelet.service
```
nit 1: I prefer always writing `curl -sSL`...
nit 2: can you just redirect it to the `/etc/systemd/system/kubelet.service` directly instead of `cd`-ing into the dir first to save a step?

luxas commented on docs/setup/independent/install-kubeadm.md:
```diff
@@ -220,9 +231,44 @@ systemctl enable kubelet && systemctl start kubelet
+curl -L "https://raw.githubusercontent.com/kubernetes/kubernetes/${RELEASE}/build/debs/kubelet.service" | sed 's:/usr/bin:/opt/bin:g' > kubelet.service
+mkdir -p "/etc/systemd/system/kubelet.service.d"
+cd "/etc/systemd/system/kubelet.service.d"
+curl -L "https://raw.githubusercontent.com/kubernetes/kubernetes/${RELEASE}/build/debs/10-kubeadm.conf" | sed 's:/usr/bin:/opt/bin:g' > 10-kubeadm.conf
```
same nit here: can you redirect to the file directly?

luxas commented on docs/setup/independent/install-kubeadm.md:
```diff
@@ -220,9 +231,44 @@ systemctl enable kubelet && systemctl start kubelet
+
+cd "/etc/systemd/system/"
+curl -L "https://raw.githubusercontent.com/kubernetes/kubernetes/${RELEASE}/build/debs/kubelet.service" | sed 's:/usr/bin:/opt/bin:g' > kubelet.service
+mkdir -p "/etc/systemd/system/kubelet.service.d"
```
no " needed here, be consistent in not using " when not needed

klausenbusk commented:
I have fixed all the nits, some comments:

> Please document what CoreOS version you used to test this with...

I have added the tested version in a parentheses, as Container Linux use a [different release model](https://coreos.com/releases/) than most Linux distro. They could break the script in `15xx.1.0` and the script probably also work with a year old version of Container Linux, so I don't think it make sense to say that we support ex: `1576.4.0+`.

> nit 1: I prefer always writing curl -sSL...

I have change all `curl` commands to that expect the `{kubeadm,kubelet,kubectl}` download, as they are "big" and I think a progress bar make sense in that case. What do you think?

klausenbusk commented:
I will squash the commits later, I'm just adding "dummy" comments as it easier to review.

luxas commented:
please squash the commits

klausenbusk commented:
> please squash the commits

Done

luxas commented:
I can also confirm that these instructions will give you a Kubernetes Conformant cluster. I tested this out manually and ran the [official e2e tests](https://github.com/cncf/k8s-conformance/tree/master/v1.8/kubeadm), and all tests passed :smile:!
